block/local/adapter.go

```diff
@@ -70,9 +70,24 @@ func NewAdapter(path string, opts ...func(a *Adapter)) (*Adapter, error) {
 	}
 	return adapter, nil
 }
+func resolveNamespace(obj block.ObjectPointer) (block.QualifiedKey, error) {
+	qualifiedKey, err := block.ResolveNamespace(obj.StorageNamespace, obj.Identifier)
+	if err != nil {
+		return qualifiedKey, err
+	}
+	if qualifiedKey.StorageType != block.StorageTypeLocal {
+		return qualifiedKey, block.ErrInvalidNamespace
+	}
+	return qualifiedKey, nil
+}
 
-func (l *Adapter) getPath(identifier string) string {
-	return path.Join(l.path, identifier)
+func (l *Adapter) getPath(identifier block.ObjectPointer) (string, error) {
+	obj, err := resolveNamespace(identifier)
+	if err != nil {
+		return "", err
+	}
+	p := fmt.Sprintf("%s/%s", obj.StorageNamespace, obj.Key)
+	return path.Join(l.path, p), nil
```

Comment:
I think this should work:
```suggestion
	return path.Join(l.path, obj.StorageNamespace, obj.Key), nil
```